processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContexts.java

```diff
@@ -250,23 +271,31 @@ public void partitionsResumed(List<TopicPartition> partitions) {
      */
     public void maybeHandlePropertyReload() {
         if (reloadRequested.get()) {
-            if (totalPendingTasks() > 0) {
-                logger.debug("Waiting pending tasks for property reload.");
-                return;
+            lock.lock();
+            try {
+                List<TopicPartition> reloadableTopicPartitions = reloadStates.entrySet()
+                        .stream()
+                        .filter(entry -> entry.getValue()
+                                         && contexts.get(entry.getKey()).pendingTasksCount() == 0)
```

Comment:
This could cause NPE because currently `reloadStates` entry never removed even when partition is removed.

To fix that, we can add a code to remove `reloadStates` entry at the same time we modify `contexts`.

However, then, we have to maintain two separate Map which holds per-partition states, which is bothersome and an error-prone.

So I think we should `reloadRequested` flag in PartitionContext itself. WDYT

Reply:
I got the point.
Having a `reloadRequested` flag in `PartitionContext` seems good.